aptos-move/framework/src/natives/string_utils.rs

```diff
@@ -0,0 +1,325 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
```

Comment:
Can we reuse functions from https://github.com/aptos-labs/aptos-core/blob/main/third_party/move/move-stdlib/src/natives/debug.rs#L250?

Reply:
I think that has quadratic complexity, and is in behind testing feature flag.

Reply:
cc @alinush What do you think?

Reply:
> Can we reuse functions from https://github.com/aptos-labs/aptos-core/blob/main/third_party/move/move-stdlib/src/natives/debug.rs#L250?

For now please not. Sorry I know this is confusing as we now have duplicates of some native functions. But we decided a while ago to branch the native functions so we can adapt their APIs smoothly for our needs as our VM evolves. All our native code should be in `aptos-move/framework`. `third_party/move` contains reference implementations which we do not touch for now.

We can discuss how we push stuff from aptos-move/framework to third_party, but I would be careful with this, as any code we develop cannot reference back into aptos-core, which restricts it a bit. We need more experience with the current setup before we do this.

Reply:
I'm surprised that there's a simpler way of doing this. My way seems hacky in comparison.

I don't understand yet how @gerben-stavenga was able to do this differently/more efficiently in this PR. I was working off of some example code that Sam (Mysten) gave me, trying to adapt it to every case.

Reply:
I figured out what is going on. One cannot use value_to_move_value with fully annotated layout, because that would panic here 

https://github.com/aptos-labs/aptos-core/blob/a1ae60836f987cacb6ed3f731e585df7c684c67e/third_party/move/move-core/types/src/value.rs#L242

This implementation is not using that function and is just traversing the value directly as it is printing. I refactored debug code.

1) debug.move is living in aptos-stdlib (not in move-stdlib) so I moved the native impl to aptos-stdlib natives impls
2) I made the debug native just print string. And use string-utils::debug_string to convert value to string